Content.Client/GameObjects/Components/Photography/PhotoCameraComponent.cs

```diff
@@ -0,0 +1,204 @@
+
+                if (!_parent._uiUpdateNeeded)
+                {
+                    return;
+                }
+
+                _parent._uiUpdateNeeded = false;
+
+                var message = new FormattedMessage();
+
+                if (_parent.CameraOn)
+                {
+                    message.AddMarkup(Loc.GetString("[color=green]On[/color]\n"));
+                }
+                else
+                {
+                    message.AddMarkup(Loc.GetString("[color=red]Off[/color]\n"));
+                }
+
+                message.AddMarkup(Loc.GetString("Film: [color={0}]{1}/{2}[/color], ",
+                    _parent.Film <= 0 ? "red" : "white", _parent.Film, _parent.FilmMax));
+                message.AddMarkup(Loc.GetString("Radius: [color=white]{0}x{0}[/color]", _parent.Radius * 2));
+
+                _label.SetMessage(message);
+            }
+        }
+
+        public override void HandleNetworkMessage(ComponentMessage message, INetChannel netChannel, ICommonSession session = null)
+        {
+            base.HandleNetworkMessage(message, netChannel, session);
+
+            switch (message)
+            {
+                case SuicideSelfieMessage selfie:
+                    var viewport = _eyeManager.GetWorldViewport();
+                    var center = _eyeManager.WorldToScreen(_playerManager.LocalPlayer.ControlledEntity.Transform.GridPosition);
+                    TryTakePhoto(selfie.Who, new Vector2(center.X, center.Y), true);
+                    break;
+            }
+        }
+
+        public async void TryTakePhoto(EntityUid author, Vector2 photoCenter, bool suicide = false)
+        {
+            if (!CameraOn)
+            {
+                _notifyManager.PopupMessageCursor(_playerManager.LocalPlayer.ControlledEntity, Loc.GetString("Turn the {0} on first!", Owner.Name));
+                return;
+            }
+
+            if(Film <= 0)
+            {
+                _notifyManager.PopupMessageCursor(_playerManager.LocalPlayer.ControlledEntity, Loc.GetString("No film!"));
+                return;
+            }
+
+            //Play sounds
+            SendNetworkMessage(new TakingPhotoMessage());
+
+            //Take a screenshot before the UI, and then crop it to the photo radius
+            var screenshot = await _clyde.ScreenshotAsync(ScreenshotType.BeforeUI);
+            var cropDimensions = EyeManager.PixelsPerMeter * (Radius * 4);
+
+            //We'll try and center, but otherwise we'll shift the box so it doesn't go outside the viewport
+            var cropX = (int)Math.Clamp(Math.Floor(photoCenter.X - cropDimensions / 2), 0, screenshot.Width - cropDimensions);
+            var cropY = (int)Math.Clamp(Math.Floor(photoCenter.Y - cropDimensions / 2), 0, screenshot.Height - cropDimensions);
+
+            Logger.InfoS("photo", $"cropX:{cropX}, cropY:{cropY}, w:{screenshot.Width}, h:{screenshot.Height}");
+
+            var success = false;
+            var filename = $"MY_PHOTO";
```

Comment:
?

Reply:
this and the next 3 comments are just identical code from taking a screenshot.
Because that's all this does, is take a screenshot, crop it, and then try and save it.

The file being saved here is the last photo you took, it's saved to disk as a PNG, and later that same PNG file's bytes are uploaded to the server (so they can be reinterpreted and resaved as a PNG on the server side, ready to serve up the photo to anyone who looks at it)